probe/topology_tagger.go

```diff
@@ -19,18 +16,19 @@ func (topologyTagger) Name() string { return "Topology" }
 
 // Tag implements Tagger
 func (topologyTagger) Tag(r report.Report) (report.Report, error) {
-	for val, topology := range map[string]*report.Topology{
+	for name, t := range map[string]*report.Topology{
 		"endpoint":        &(r.Endpoint),
 		"address":         &(r.Address),
 		"process":         &(r.Process),
 		"container":       &(r.Container),
 		"container_image": &(r.ContainerImage),
+		"pod":             &(r.Pod),
+		"service":         &(r.Service),
 		"host":            &(r.Host),
 		"overlay":         &(r.Overlay),
 	} {
-		metadata := map[string]string{Topology: val}
-		for id, node := range topology.Nodes {
-			topology.AddNode(id, node.WithMetadata(metadata))
+		for id, node := range t.Nodes {
+			t.AddNode(id, node.WithID(id).WithTopology(name))
```

Comment:
I think I'd much prefer it if AddNode only took one argument, and the node constructors were forced to pass an ID (instead of the ID getting filled out after-the-fact).